src/contracts/core/DelegationManager.sol

```diff
@@ -442,6 +442,60 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         }
     }
 
+    /**
+     * @notice Called by AVSs to register an operator with the AVS.
+     * @param operator The address of the operator to register.
+     * @param signatureWithSaltAndExpiry The signature, salt, and expiry of the operator's signature.
+     */
+    function registerOperatorWithAVS(address operator, ISignatureUtils.SignatureWithSaltAndExpiry memory signatureWithSaltAndExpiry) external {
+        require(signatureWithSaltAndExpiry.expiry > block.timestamp, "DelegationManager.registerOperatorWithAVS: signature expired");
+        require(operatorSaltIsSpent[operator][signatureWithSaltAndExpiry.salt] == false, "DelegationManager.registerOperatorWithAVS: salt already used");
+        require(!registeredWithAVS[msg.sender][operator], "DelegationManager.registerOperatorWithAVS: operator already registered");
+
+        // Calculate the digest hash
+        bytes32 operatorRegistrationDigestHash = calculateOperatorRegistrationDigestHash({
+            operator: operator, 
+            avs: msg.sender, 
+            salt: signatureWithSaltAndExpiry.salt, 
+            expiry: signatureWithSaltAndExpiry.expiry
+        });
+
+        // Check that the signature is valid
+        EIP1271SignatureUtils.checkSignature_EIP1271(
+            operator,
+            operatorRegistrationDigestHash,
+            signatureWithSaltAndExpiry.signature
+        );
+
+        // mark the salt as spent
+        operatorSaltIsSpent[operator][signatureWithSaltAndExpiry.salt] = true;
+        // register operator
+        registeredWithAVS[msg.sender][operator] = true;
+
+        emit OperatorRegistrationStatusUpdatedForAVS(operator, msg.sender, OperatorRegistrationStatus.REGISTERED);
+    }
+
+    /**
+     * @notice Called by AVSs to deregister an operator with the AVS.
+     * @param operator The address of the operator to deregister.
+     */
+    function deregisterOperatorFromAVS(address operator) external {
```

Comment:
Questions:
1. Are we ok with this being unsigned?
2. Do we want the operator to be able to deregister themselves, or should this only be something the AVS does?

Reply:
I _personally think_:

- for (1) yes, the AVS should be able to unilaterally do this
- we should aspire to also accomplish (2), but I'm unclear on if this is required (as part of the upcoming release, at least). will flag this to people for input

Reply:
I suspect it's just fine to keep (2) as aspirational for the moment, since it's a significant expansion of scope with unclear need (we then have to figure out how to communicate this deregistration to AVSs, at least).

Reply:
1. Agree with this. Also note that AVS' can eject operators, so it makes sense that an AVS can force deregister an operator
2. #2 doesn't seem necessary in the short term and adds complexity

Reply:
> What do you guys mean by "AVS can eject operators"? Is that based on the contracts in this repo? Or is this a general requirement that we have? Also what's the difference between ejecting and deregistering? Is "deregistering" just the name you use to refer to the registry in the DM?

I think we use the term "ejection" to more specifically mean an AVS deregistering an operator _without the operator initiating the action_. While not a strict requirement, I would say AVSs SHOULD have some kind of ejection mechanism like what's in this repo

> Is there a difference between deregistering and just updating my stake to be 0? Seems like if operators can't either deregister, or remove their stake from an AVS, then they are stuck there forever..?

ideally, EigenLayer would know + index when an AVS no longer considers an operator to be opted-into/serving the service.
I think aspirationally there would be a high level of agreement/continuous consent in this relationship, where either party can sever the relationship (subject to some rules).
I agree right now it feels like operators could kind of "get stuck" in AVSs, _but_ I'm not seeing specific downsides to that, so I feel OK deferring development of some kind of operator escape hatch mechanism until later.